cpp/oneapi/dal/backend/interop/error_convertor.hpp

```diff
@@ -0,0 +1,366 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+
+#include "oneapi/dal/exception.hpp"
+#include "daal/include/services/error_handling.h"
+#include "daal/include/services/internal/status_to_error_id.h"
+
+namespace oneapi::dal::backend::interop {
+
+void status_to_exception_default(const daal::services::Status &s) {
+    using namespace daal::services;
+    using namespace daal::services::internal;
+
+    const ErrorID error = get_error_id(s);
+    const char * description = s.getDescription();
+
+    switch (error) {
```

Comment:
Maybe:
```c++
const char * description = s.getDescription();
if (!error ) {
  throw invalid_argument(description);
}
```

Reply:
There are different types of error classes.